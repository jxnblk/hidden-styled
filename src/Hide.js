import React from 'react'
import styled from 'styled-components'
import { display } from 'styled-system'

const withHideProps = Component => ({
  xs,
  sm,
  md,
  lg,
  xl,
  display,
  ...props
}) =>
  <Component
    {...props}
    display={display || [
      xs,
      sm,
      md,
      lg,
      xl
    ].map(n => n ? 'none' : 'block')}
  />

const Hide = styled(
  withHideProps(
    styled(({ display, ...props }) =>
      <div {...props} />
    )([], display)
  )
)([])

export default Hide
