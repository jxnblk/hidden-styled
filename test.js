import test from 'ava'
import React from 'react'
import { ThemeProvider } from 'styled-components'
import { createElement as h } from 'react'
import { create as render } from 'react-test-renderer'
import Hide from './src/Hide'

test('renders', t => {
  const json = render(h(Hide)).toJSON()
  t.snapshot(json)
})

test('renders with xs', t => {
  const json = render(h(Hide, { xs: true })).toJSON()
  t.snapshot(json)
})

test('renders with sm', t => {
  const json = render(h(Hide, { sm: true })).toJSON()
  t.snapshot(json)
})

test('renders with md', t => {
  const json = render(h(Hide, { md: true })).toJSON()
  t.snapshot(json)
})

test('renders with lg', t => {
  const json = render(h(Hide, { lg: true })).toJSON()
  t.snapshot(json)
})

test('breakpoints can be configured with theme', t => {
  const json = render(
    <ThemeProvider theme={{
      breakpoints: [
        24,
        36,
        48,
        96
      ]
    }}>
      <Hide xs />
    </ThemeProvider>
  ).toJSON()
  t.snapshot(json)
})
