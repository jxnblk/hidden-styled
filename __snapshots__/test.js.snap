// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breakpoints can be configured with theme 1`] = `
<div
  className="sc-bwzfXH uTxCW sc-bdVaJa lkmfnQ"
/>
`;

exports[`renders 1`] = `
<div
  className="sc-bwzfXH uTxCW sc-bdVaJa iQKmFe"
/>
`;

exports[`renders with lg 1`] = `
<div
  className="sc-bwzfXH uTxCW sc-bdVaJa kHiLOl"
/>
`;

exports[`renders with md 1`] = `
<div
  className="sc-bwzfXH uTxCW sc-bdVaJa beuNRA"
/>
`;

exports[`renders with sm 1`] = `
<div
  className="sc-bwzfXH uTxCW sc-bdVaJa brqKIn"
/>
`;

exports[`renders with xs 1`] = `
<div
  className="sc-bwzfXH uTxCW sc-bdVaJa iQKmFe"
/>
`;
